eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,778 @@
+        }
+
+
+        inline double sigma_eetomumu(const double & E)
+        {
+            // Conversion factor between GeV^2 and nb
+            const double speedoflight = 299792458.; // Exact value
+            const double GeVtonb = 10 * power_of<2>( 1.e18 * hbar * speedoflight);
+
+            return GeVtonb * 4.0 * M_PI * alpha_em * alpha_em / (3.0 * E*E);
+        }
+
+        // amplitude of ee -> channel
+        complex<double> amplitude_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Conversion factor between GeV^2 and nb
+            const double speedoflight = 299792458.; //Exact value
+            const double GeVtonb = 10 * power_of<2>( 1.e18 * hbar * speedoflight);
+
+            // Channel properties
+            const double Nf = 2 * K->_channels[Channels(channel)]->_l_orbital + 1;
+            const double rhof = std::abs(K->_channels[Channels(channel)]->rho(intermediate_result->s));
+
+            // Get T-matrix[ee, channel]
+            const complex<double> T1f = intermediate_result->tmatrix_row_0[Channels(channel)];
+
+            return sqrt(GeVtonb / intermediate_result->s * Nf * rhof)* T1f;
+        }
+
+        complex<double> T_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Get T-matrix[ee, channel]
+            const complex<double> T1f = intermediate_result->tmatrix_row_0[Channels(channel)];
+
+            return T1f;
+        }
+
+        complex<double> T2_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            // Get T-matrix[ee, channel] on the second RS
+            const complex<double> T1f = intermediate_result->tmatrix2_row_0[Channels(channel)];
+
+            return T1f;
+        }
+
+        double sigma_eetochannel(const IntermediateResult * intermediate_result, const Channels & channel)
+        {
+            return norm(amplitude_eetochannel(intermediate_result, channel));
+        }
+
+        // Widths
+        double res_partial_width(const Resonances & resonance, const Channels & channel)
+        {
+            return K->partial_width(resonance, channel);
+        }
+
+        double res_total_width(const Resonances & resonance)
+        {
+            return K->width(resonance);
+        }
+
+        // Ratios
+        double R(const IntermediateResult * intermediate_result)
+        {
+            double total_xsec = 0.0;
+
+            for (unsigned i = 1; i < EEToCCBar::nchannels; i++)
+            {
+                total_xsec += sigma_eetochannel(intermediate_result, Channels(i));
+            }
+
+            return total_xsec / sigma_eetomumu(abs(intermediate_result->E)) + Rconstant; // Add constant term
+        }
+
+        // Spectral function
+        double spectral_function(const double E, const Resonances & resonance)
+        {
+            return K->spectral_function(resonance, E*E);
```

Comment:
Coding style: whitespace left and right of binary operator (throughout the file).